CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -72,19 +72,42 @@ namespace CefSharp
             CefRefPtr<CefClient>& client, CefBrowserSettings& settings, bool* no_javascript_access)
         {
             ILifeSpanHandler^ handler = _browserControl->LifeSpanHandler;
+			IWebBrowser^ newBrowser = nullptr;
 
             if (handler == nullptr)
             {
                 return false;
             }
 
-            return handler->OnBeforePopup(_browserControl, gcnew CefFrameWrapper(frame, _browserControl->BrowserAdapter), StringUtils::ToClr(target_url),
-                windowInfo.x, windowInfo.y, windowInfo.width, windowInfo.height, *no_javascript_access);
+            bool returnVal = handler->OnBeforePopup(_browserControl, gcnew CefFrameWrapper(frame, _browserControl->BrowserAdapter), StringUtils::ToClr(target_url),
+				windowInfo.x, windowInfo.y, windowInfo.width, windowInfo.height, *no_javascript_access, newBrowser);
+
+			IWebBrowserInternal^ newBrowserInternal = dynamic_cast<IWebBrowserInternal^>(newBrowser);
+
+			if (newBrowserInternal != nullptr)
+			{
+
+				IRenderWebBrowser^ renderBrowser = dynamic_cast<IRenderWebBrowser^>(newBrowser);
+				if (renderBrowser != nullptr)
+				{
+					windowInfo.SetAsWindowless(windowInfo.parent_window, TRUE);
+				}
+
+				ManagedCefBrowserAdapter^ browserAdapter = dynamic_cast<ManagedCefBrowserAdapter^>(newBrowserInternal->BrowserAdapter);
+				if (browserAdapter != nullptr)
+				{			
+					client = browserAdapter->GetClientAdapter().get();
+				}
+
+			}
+
+			return returnVal;
+
         }
 
         void ClientAdapter::OnAfterCreated(CefRefPtr<CefBrowser> browser)
-        {
-            if (browser->IsPopup())
+        {			
+            if (browser->IsPopup() && !IsOffscreen())
```

Comment:
The events are raised against the created instance (the popup). As it's now being displayed as windowless, we can't treat it as a popup.